### DISCUSSION TITLE
Codemagic "Installing SDKs" for Flutter timing out

### DISCUSSION THREAD

**Discussion by kieranmccwayleadr** [codemagic-ci-cd/codemagic-docs#2425] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Create a build for your Flutter appplication using Flutter 3.16.2

### Expected results

The SDK should be downloaded in the expected time scale and continue the build cycle.

### Actual results

We continuously get timeouts yet the download is getting retried:

```== Use Flutter version 3.16.2 ==
> curl --fail --show-error --silent --output flutter_macos_arm64_3.16.2-stable.zip --connect-timeout 10.00 --max-time 45.00 https://storage.googleapis.com/flutter_infra_release/releases/stable/macos/flutter_macos_arm64_3.16.2-stable.zip
curl: (28) Operation timed out after 45000 milliseconds with 59032868 out of 1142143711 bytes received

Downloading Flutter release timed out, try again...
> curl --fail --show-error --silent --output flutter_macos_arm64_3.16.2-stable.zip --connect-timeout 10.00 --max-time 45.00 https://storage.googleapis.com/flutter_infra_release/releases/stable/macos/flutter_macos_arm64_3.16.2-stable.zip
curl: (28) Operation timed out after 45002 milliseconds with 176686174 out of 1142143711 bytes received

Downloading Flutter release timed out, try again...
> curl --fail --show-error --silent --output flutter_macos_arm64_3.16.2-stable.zip --connect-timeout 10.00 --max-time 45.00 https://storage.googleapis.com/flutter_infra_release/releases/stable/macos/flutter_macos_arm64_3.16.2-stable.zip
curl: (28) Operation timed out after 45000 milliseconds with 569823164 out of 1142143711 bytes received

Downloading Flutter release timed out

Build failed :|
Downloading Flutter release timed out```

### Build id (optional)

65fc48d518c24e59ef3ff38e

**Comment by fomenkodm93** [comment]
I have the same problem with 6 builds in order.
build numbers: 65fc4b9b0e81077cf17eed88, 65fc4da2b214bae3d58179ae, 65fc4a50ab357296da96a724, 65fc48bc0e81077cf17eebc2, 65fc434355e9c53748410fa7, 65fc3cfb3d3c446aebefbf50

Can't build anything but need it urgently!

**Comment by markbreuss** [comment]
Same issue here - build uid: 65fc4f1556cf07dc6b9fc121

**Comment by kalyujniy** [comment]
With Flutter 3.19.3 the same issue
`
== Use Flutter channel stable ==
> curl --fail --show-error --silent --output flutter_macos_arm64_3.19.3-stable.zip --connect-timeout 10.00 --max-time 45.00 https://storage.googleapis.com/flutter_infra_release/releases/stable/macos/flutter_macos_arm64_3.19.3-stable.zip
curl: (28) Operation timed out after 45007 milliseconds with 56098409 out of 1167817719 bytes received

Downloading Flutter release timed out, try again...
> curl --fail --show-error --silent --output flutter_macos_arm64_3.19.3-stable.zip --connect-timeout 10.00 --max-time 45.00 https://storage.googleapis.com/flutter_infra_release/releases/stable/macos/flutter_macos_arm64_3.19.3-stable.zip
curl: (28) Operation timed out after 45007 milliseconds with 20208912 out of 1167817719 bytes received

Downloading Flutter release timed out, try again...
> curl --fail --show-error --silent --output flutter_macos_arm64_3.19.3-stable.zip --connect-timeout 10.00 --max-time 45.00 https://storage.googleapis.com/flutter_infra_release/releases/stable/macos/flutter_macos_arm64_3.19.3-stable.zip
curl: (28) Operation timed out after 45006 milliseconds with 115198820 out of 1167817719 bytes received

Downloading Flutter release timed out

Build failed :|
Downloading Flutter release timed out`

**Comment by mikhail-tokarev** [comment]
We are seeing an increased number of network errors when interacting with Google Cloud Storage and our team is looking how we can improve things. 

I'll close this to keep one ticket for further communications https://github.com/orgs/codemagic-ci-cd/discussions/2424
